Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -146,22 +164,26 @@ extension HTTPClient {
                 throw HTTPClientError.emptyScheme
             }
 
-            guard Request.isSchemeSupported(scheme: scheme) else {
-                throw HTTPClientError.unsupportedScheme(scheme)
-            }
+            if Kind.host.isSchemeSupported(scheme: scheme) {
+                self.kind = .host
+                guard let host = url.host else {
+                    throw HTTPClientError.emptyHost
+                }
 
-            guard let host = url.host else {
-                throw HTTPClientError.emptyHost
+                self.host = host
+            } else if Kind.unixSocket.isSchemeSupported(scheme: scheme) {
+                self.kind = .unixSocket
+                self.host = ""
+            } else {
+                throw HTTPClientError.unsupportedScheme(scheme)
             }
```

Comment:
This block can be rewritten to be substantially cleaner by defining some helpers:

```swift
extension Kind {
    private static var hostSchemes = ["http", "https"]
    private static var unixSchemes = ["unix"]

    init(forScheme scheme: String) throws {
        if Kind.hostSchemes.contains(scheme) {
            self = .host
        } else if Kind.unixSchemes.contains(scheme) {
            self = .unixSocket
        } else {
            throw HTTPClientError.unsupportedScheme
        }
    }

    func hostFromURL(_ url: URL) throws -> String {
        switch self {
        case .host:
            guard let host = url.host else {
                throw HTTPClientError.emptyHost
            }
            return host
        case .unixSocket:
            return ""
        }
    }
}
```

This code then becomes:

```swift
self.kind = try Kind(scheme: scheme)
self.host = try self.kind.hostFromURL(url)
```

We can then also rewrite `kind.isSchemeSupported` in terms of our statics. This should put most of the complexity into the enum definition, which helps keep the code elsewhere a lot nicer.

Reply:
thanks for the feedback. Applied as requested.